core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2541,6 +2541,15 @@ private[spark] object Utils extends Logging {
     master == "local" || master.startsWith("local[")
   }
 
+  /**
+   * Push based shuffle can only be enabled when external shuffle service is enabled.
+   * In the initial version, we cannot support pushed based shuffle and adaptive execution
+   * at the same time. Will improve this in a later version.
```

Comment:
Comment no longer valid. We can support push-based shuffle and AQE together now.

Reply:
Fixed